tests/unit/test-cmd/test.build.js

```diff
@@ -37,6 +63,103 @@ describe('build', () => {
+    return withTempDir(
+      (tmpDir) =>
+        build({
+          sourceDir: fixturePath('minimal-localizable-web-ext'),
+          artifactsDir: tmpDir.path(),
+        })
+        .then((buildResult) => {
+          assert.match(buildResult.extensionPath,
+                       /name_of_the_extension-1\.0\.zip$/);
+          return buildResult.extensionPath;
+        })
+    );
+  });
+
+  it('gives the correct name to a localized' +
+    'extension with repeating pattern in manifest', () => {
+    return withTempDir(
+      (tmpDir) => {
+        const messageFileName = path.join(tmpDir.path(), 'messages.json');
+        //This is missing the 'message' key
+        fs.writeFileSync(messageFileName,
+          `{"extensionName": {
+              "message": "example extension",
+              "description": "example extension"
+            }
+          }`);
+        return getDefaultLocalizedName({
+          messageFile: messageFileName,
+          manifestData: basicLocalizedManifestWithRepeatingPattern,
+        })
+          .then((result) => {
+            assert.match(result, /example extensionexample extension/);
+          });
+      }
+    );
+  });
+
+  it('checks locale file for malformed json', () => {
+    return withTempDir(
+      (tmpDir) => {
+        const messageFileName = path.join(tmpDir.path(), 'messages.json');
+        fs.writeFileSync(messageFileName,
+          '{"simulated:" "json syntax error"');
+        return getDefaultLocalizedName({
+          messageFile: messageFileName,
+          manifestData: manifestWithoutApps,
+        })
+          .then(makeSureItFails())
+          .catch((error) => {
+            assert.instanceOf(error, WebExtError);
+            assert.match(
+              error.message,
+              /The JSON file is malformed: SyntaxError: Unexpected string/);
+          });
+      }
+    );
+  });
+
+  it('checks locale file for incorrect format', () => {
+    return withTempDir(
+      (tmpDir) => {
+        const messageFileName = path.join(tmpDir.path(), 'messages.json');
+        //This is missing the 'message' key
+        fs.writeFileSync(messageFileName,
+          `{"extensionName": {
+              "description": "example extension"
+              }
+          }`);
+        return getDefaultLocalizedName({
+          messageFile: messageFileName,
+          manifestData: basicLocalizedManifest,
+        })
+          .then(makeSureItFails())
+          .catch((error) => {
+            assert.instanceOf(error, WebExtError);
+            assert.match(
+              error.message,
+              /The locale file .*messages\.json .* key: extensionName/);
+          });
+      }
+    );
+  });
+
+  it('throws an error if the locale file doesn\'t exist', () => {
+    return getDefaultLocalizedName({
+      messageFile: '/path/to/non-existent-dir/messages.json',
+      manifestData: manifestWithoutApps,
+    })
+    .then(makeSureItFails())
```

Comment:
this needs to be indented two spaces to show that it's a continuation of the call to `getDefaultLocalizedName()`